scylla/src/transport/load_balancing/default.rs

```diff
@@ -70,13 +70,14 @@ enum StatementType {
 /// It can be configured to be datacenter-aware and token-aware.
 /// Datacenter failover for queries with non local consistency mode is also supported.
 /// Latency awareness is available, althrough not recommended.
+#[allow(clippy::type_complexity)]
 pub struct DefaultPolicy {
     preferences: NodeLocationPreference,
     is_token_aware: bool,
     permit_dc_failover: bool,
-    pick_predicate: Box<dyn Fn(&NodeRef) -> bool + Send + Sync>,
+    pick_predicate: Box<dyn Fn(&(NodeRef<'_>, Shard)) -> bool + Send + Sync>,
     latency_awareness: Option<LatencyAwareness>,
-    fixed_shuffle_seed: Option<u64>,
+    fixed_seed: Option<u64>,
```

Comment:
Justification for this name change: the seed is now not only used for **shuffling** nodes, but also for sampling a random shard in case we don't know the proper one.